### PR TITLE
feat: inputNumber support customize upIcon/downIcon

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/33641
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   feat: inputNumber support customize upIcon/downIcon        |
| 🇨🇳 Chinese |  功能:  `inputNumber `组件支持自定义 `向上icon`  `向下icon`       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/input-number/demo/controls.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+order: 1
+title:
+  zh-CN: 图标按钮
+  en-US: Icon
+---
+
+## zh-CN
+
+当需要使用自定义`Icon`时，可以设置  `Icon` 组件为 `upIcon` 和 `downIcon` 的属性值
```
Comment (afc163) [inline components/input-number/demo/controls.md:10]:
```suggestion
可以扩展 `controls` 属性用以设置自定义图标。
```

File: components/input-number/demo/controls.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+When you need to use a custom `Icon`, you can set the `Icon` component as the property value of `upIcon` and `downIcon`
+
+```jsx
+import { InputNumber } from 'antd';
+import { ArrowUpOutlined,ArrowDownOutlined } from '@ant-design/icons';
+function onChange(value) {
+  console.log('changed', value);
+}
+
+ReactDOM.render(<InputNumber min={1} max={10} controls={{ upIcon: <ArrowUpOutlined />, downIcon: <ArrowDownOutlined/>} } defaultValue={3} onChange={onChange} />, mountNode);
```
Comment (afc163) [inline components/input-number/demo/controls.md:23]:
```suggestion
import { InputNumber } from 'antd';
import { ArrowUpOutlined, ArrowDownOutlined } from '@ant-design/icons';

ReactDOM.render(
  <InputNumber
   controls={{ upIcon: <ArrowUpOutlined />, downIcon: <ArrowDownOutlined/>} }
  />
, mountNode);
```

File: components/input-number/index.tsx [hunk 3]
```diff
@@ -37,12 +38,22 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
...
     ...others
   } = props;
 
   const prefixCls = getPrefixCls('input-number', customizePrefixCls);
-  const upIcon = <UpOutlined className={`${prefixCls}-handler-up-inner`} />;
-  const downIcon = <DownOutlined className={`${prefixCls}-handler-down-inner`} />;
+  let upIcon = <UpOutlined className={`${prefixCls}-handler-up-inner`} />
+  let downIcon = <DownOutlined className={`${prefixCls}-handler-down-inner`} />
+
+  if(typeof controls === 'object'){
```
Comment (afc163) [inline components/input-number/index.tsx:49]:
```suggestion
  if(typeof controls === 'object') {
```

File: components/input-number/index.tsx [hunk 4]
```diff
@@ -37,12 +38,22 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
...
+    upIcon = controls.upIcon ? (
```
Comment (afc163) [inline components/input-number/index.tsx:50]:
```suggestion
    upIcon = typeof controls.upIcon === 'undefined' ? (
```

File: components/input-number/index.zh-CN.md [hunk 5]
```diff
@@ -22,7 +22,7 @@ cover: https://gw.alipayobjects.com/zos/alicdn/XOS8qZ0kU/InputNumber.svg
 | addonBefore | 带标签的 input，设置前置标签 | ReactNode | - | 4.17.0 |
 | autoFocus | 自动获取焦点 | boolean | false | - |
 | bordered | 是否有边框 | boolean | true | 4.12.0 |
-| controls | 是否显示增减按钮 | boolean | true | 4.17.0 |
+| controls | 是否显示增减按钮,设置自定义箭头`icon` | boolean \| { upIcon?: React.ReactNode; downIcon?: React.ReactNode; } | - | 4.17.0 |
```
Comment (afc163) [inline components/input-number/index.zh-CN.md:25]:
```suggestion
| controls | 是否显示增减按钮，也可设置自定义箭头图标 | boolean \| { upIcon?: React.ReactNode; downIcon?: React.ReactNode; } | - | 4.17.0 |
```

**Comment by afc163** [comment]
图片

这个 commit 不属于这个 PR。

**Comment by afc163** [comment]
图片

代码冲突了。

**Comment by heiyu4585** [comment]
> 图片
> 
> 代码冲突了。

收到,我处理下

**Pull request closed without merging**